platform/darwin/src/MGLConversionValue.h

```diff
@@ -0,0 +1,73 @@
+//
+//  MGLConversionValue.h
+//  ios
+//
+//  Created by Anand Thakker on 2/10/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+#ifndef MGLConversionValue_h
+#define MGLConversionValue_h
+
+class MGLConversionValue {
+public:
+    Value(NSObject * _value) : value(_value) {}
+    
+    bool isNull() const {
+        return value;
+    }
+    
+    bool isArray() const {
+        return [value isKindOfClass:[NSArray class]];
+    }
+    
+    bool isObject() const {
+        return [value isKindOfClass:[NSDictionary class]];
+    }
+    
+    bool isString() const {
+        return [value isKindOfClass:[NSString class]];
+    }
+    
+    bool isBool() const {
+        return [value isKindOfClass:[NSNumber class]] &&
+                ((strcmp([value objCType], @encode(char)) == 0) ||
+                (strcmp([value objCType], @encode(BOOL)) == 0));
+    }
+    
+    bool isNumber() const {
+        return [value isKindOfClass:[NSNumber class]] && !isBool();
+    }
+    
+    std::string toString() const {
+        return std::string([(NSString *)value UTF8String]);
+    };
+    
+    float toNumber() const {
```

Comment:
@1ec5 hmm, yeah -- so the reason I went with this here is that (I believe) the mbgl conversion system expects `mbgl::style::conversion::toNumber` to return `optional<float>`. I suppose we could have `MGLConversionValue::toNumber()` return an NSNumber, and leave it to `conversion::toNumber()` to retrieve the floatValue.  (Though this makes me second-guess whether the `MGLConversionValue` wrapper is even necessary -- couldn't we just overload `conversion::toNumber()` for `NSObject` directly?)

By the way, is replacing `mgl_constantMBGLValue` something I should consider as a requirement/use case for MGLConversionValue?   If so, I may need a little help understanding the function/purpose of `mgl_constantMBGLValue` and friends.